cmd/clusterctl/client/cluster/topology.go

```diff
@@ -475,7 +497,7 @@ func getTemplates(objs []*unstructured.Unstructured) []*unstructured.Unstructure
 // generateCRDs creates mock CRD objects for all the provider specific objects in the input.
 // These CRD objects will be added to the dry run client for UpdateReferenceAPIContract
 // to work as expected.
-func (t *topologyClient) generateCRDs(in *DryRunInput) []*apiextensionsv1.CustomResourceDefinition {
+func (t *topologyClient) generateCRDs(in *TopologyPlanInput) []*apiextensionsv1.CustomResourceDefinition {
```

Comment:
Can / should we maybe just pass in the objs here?

Just feels strange to me passing the TopologyPlanInput in those funcs if we only need 1 field.